c8/request.py

```diff
@@ -76,8 +76,9 @@ def __init__(
         self.headers = headers or {}
 
         # Insert default headers.
-        self.headers["content-type"] = "application/json"
-        self.headers["charset"] = "utf-8"
+        if headers is None:
+            self.headers["content-type"] = "application/json"
+            self.headers["charset"] = "utf-8"
```

Comment:
Did we test this out for all endpoints.. like all the tests are passing? since we are using headers in other endpoints too so it could probably mean that content-type won't be specified for them

Reply:
How about being more specific like if self.headers does not have "content-type" key then we set it to application/json? if you do headers is None then I think there are places were we have defined headers but not the content-type.. so the if condition being specific to content-type would be better no?

Reply:
Good point, thanks!